src/test/java/seedu/edulink/logic/commands/FilterCommandTest.java

```diff
@@ -0,0 +1,142 @@
+
+        FilterCommand filterFirstCommand = new FilterCommand(firstPredicate);
+        FilterCommand filterSecondCommand = new FilterCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(filterFirstCommand.equals(filterFirstCommand));
+
+        // same values -> returns true
+        FilterCommand filterFirstCommandCopy = new FilterCommand(firstPredicate);
+        assertTrue(filterFirstCommand.equals(filterFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(filterFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(filterFirstCommand.equals(null));
+
+        // different name -> returns false
+        assertFalse(filterFirstCommand.equals(filterSecondCommand));
+    }
+
+    @Test
+    public void execute_nonExistentTags_noPersonFound() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 0);
+
+        Tag testTag1 = new Tag("Intelligent");
+        Tag testTag2 = new Tag("Smart");
+        Set<Tag> queryTags = new HashSet<>();
+        queryTags.add(testTag1);
+        queryTags.add(testTag2);
+
+        TagsContainQueryTagsPredicate predicate = new TagsContainQueryTagsPredicate(queryTags);
+        FilterCommand command = new FilterCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_existentTag_onePersonFound() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 1);
+
+        Tag testTag1 = new Tag("TA");
+        Tag testTag2 = new Tag("Smart");
+        Set<Tag> queryTags = new HashSet<>();
+        queryTags.add(testTag1);
+        queryTags.add(testTag2);
+
+        TagsContainQueryTagsPredicate predicate = new TagsContainQueryTagsPredicate(queryTags);
+        FilterCommand command = new FilterCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE), model.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_existentTags_multiplePersonsFound() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 3);
+
+        Tag testTag1 = new Tag("friends");
+        Set<Tag> queryTags = new HashSet<>();
+        queryTags.add(testTag1);
+
+        TagsContainQueryTagsPredicate predicate = new TagsContainQueryTagsPredicate(queryTags);
+        FilterCommand command = new FilterCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE, BENSON, DANIEL), model.getFilteredPersonList());
+    }
```

Comment:
Should we also include some failed cases instead of only successful cases?

Reply:
I believe it is covered in the test function name execute_nonExistentTags_noPersonFound().